packages/main/src/MenuItem.ts

```diff
@@ -253,24 +298,86 @@ class MenuItem extends ListItem implements IMenuItem {
 		return false;
 	}
 
+	get isGroup(): boolean {
+		return false;
+	}
+
+	get _parentGroup() {
+		const parent = this.parentElement;
+		return parent && "isGroup" in parent && parent.isGroup ? parent as MenuItemGroup : null;
```

Comment:
traversing backward the DOM is wrong. more correct will be if the ui5-menu-item fires event to notify ui5-menu-item-group that change in selection was done then the selection to be handled within the parent element that has the full context.

Reply:
Refactored.